R/stackingutilities.R

```diff
@@ -492,9 +492,241 @@ getMultiOutcomeNames <- function(data)
+                                                            formula = NULL, 
+                                                            interaction = NULL, 
+                                                            subset = subset, 
+                                                            weights = weights)
+            colnames(stacking$data) <- vapply(stacking$data, 
+                                            FUN = function (x) attr(x, "label"), 
+                                            FUN.VALUE = character(1))
+            text <- stacking$data[, 1]
+            existing <- stacking$data[, 2:length(stacking$data)]
+            colnames(existing) <- TidyLabels(colnames(existing))
+            input.data <- list(text = text,
+                                existing = existing,
+                                subset = stacking$subset,
+                                weights = stacking$weights,
+                                inds = rownames(stacking$data))
+        } else if (attr(existing.categorization, "questiontype") == "PickOneMulti") {
+            # Reorder the variables in the categorization to match those
+            # of the text.
+            m <- match(colnames(existing.categorization), colnames(text))
+            existing.categorization <- existing.categorization[, m]
+            st <- stack(lapply(text, as.vector))
+            inds <- paste0(rownames(st), ".", st[, "ind"])
+            input.data <- list(text = st[, "values"],
+                                existing = unlist(existing.categorization),
+                                subset = if (length(subset) == 1) subset else rep(subset, times = n.text.vars),
+                                weights = rep(weights, times = n.text.vars),
+                                inds = inds)
+        }
+    }
+    attr(input.data$text, "label") <- text.label
+    input.data
+}
+
+# This function tries to modify the labels of a collection of
+# text variables, or variables within a TextMulti set to match
+# those of a PickOneMulti or PickAnyGrid.
+# The reason for its existence is that there are common label
+# tidying patterns that are applied to grids in Displayr
+# which mean that text labels may not ititially match. That is
+# labels on the code frames of grid questions commonly have
+# aditional common prefix and suffix text removed.
+prepareTextVariableLabelsForStackingWithGrids <- function(text, categorical.question) {
+    categorical.codeframe.labels <- names(attr(categorical.question, "codeframe"))
+    categorical.secondary.codeframe.labels <- names(attr(categorical.question, "secondarycodeframe"))
+    text.labels <- colnames(text)
+
+    if (!any(text.labels %in% categorical.codeframe.labels) && !any(text.labels %in% categorical.secondary.codeframe.labels)) {
+        # No matches for the text labels anywhere
+        # Remove common prefixes
+        text.labels <- flipFormat::TidyLabels(text.labels)
```

Comment:
Ideally remove the `flipFormat::` and add `@importFrom flipFormat TidyLabels` in the `roxygen`